src/utils/index.js

```diff
@@ -316,6 +317,29 @@ function padLeft(str, char, size) {
   }
 }
 
+export function parseURL(url) {
```

Comment:
This function doesn't have any external dependency so it should be easy to test. Could we add some to ensure this functions works as expected?